### DISCUSSION TITLE
Why react over vue

### DISCUSSION THREAD

**Discussion by DomiiBunn** [pterodactyl/panel#2868] [General Community]
Hoi! I remember at the begining of 1.0(0.8 at the time) the main idea was a rewrite in vue. What happend for the idea to be replaced with react?

P.S. Add an other category here?

**Comment by schrej** [comment]
That was mostly discussed in internal channels, but the summary is:

@DaneEveritt used React at work, and he initially wanted to learn about other frameworks, so he chose Vue
he then realized that he might prefer React over Vue, and subsequently switched to it:

- better typescript support
- easier to get started
- nicer and declarative API
- there wasn't much progress with Vue so far
- switching between Vue and React constantly was a pain
- I also advocated for React, but haven't really contributed anything to the Frontend myself so far unfortunately

**Reply by DomiiBunn** [reply]
Hmm. I personally would disagree with some points😂...the framework wars continue lol

But yeah i guess makes sense :)

**Reply by schrej** [reply]
Both also evolved since the choice was made, so some arguments might not be true anymore, e.g. TypeScript support.

**Reply by DaneEveritt** [reply]
Yeah, Vue 2 at least didn't have phenomenal typescript support, and it just felt difficult for me to wrap my head around the huge API surface with it (and more difficult without being able to have type checking).

I think it is a great framework, just wasn't a good fit for me and the project. When you only have a few hours a week at times it is really beneficial to just go with a stack you use elsewhere so there's minimal context switching necessary. It also helped me learn new things to bring to work, and I was able to bring concepts and fixes from work into the project as well. So mutually beneficial.

**Reply by DomiiBunn** [reply]
Makes sense. The for the answer(tbh idk how to mark it as answerd...no button on mobile it seams)

**Reply by schrej** [reply]
It depends on the category whether a answer can be selected. In this one it's not possible, as it's not really necessary I guess.
